packages/api-derive/src/index.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017-2019 @polkadot/ui-react-rx authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import ApiRx from '@polkadot/api/rx';
+
+import { cache } from './cache';
+import * as chain from './chain';
+import * as session from './session';
+
+// Put all derived functions in an object, for easier Object.keys()-ing.
+const functions = { chain, session };
+
+/**
+ * T represents the section here (chain, balances...), and P represents
+ * the function name (bestNumber, sessionProgress...).
+ */
+type ReturnTypes<T extends Record<keyof T, (...args: any[]) => any>> = {
+  [P in keyof T]: ReturnType<T[P]>
+};
+
+export interface Derive {
+  chain: ReturnTypes<typeof chain>;
+  session: ReturnTypes<typeof session>;
+}
+
+export default function decorateDerive (api: ApiRx): Derive {
+  const derive: Partial<Derive> = {};
```

Comment:
This could just be `const derive = {} as Derive` (and would allow you to get rid of the cast on return)